app/src/examples/SharedElementTransitions/CustomTransition.tsx

```diff
@@ -6,21 +6,41 @@ import {
   NativeStackScreenProps,
 } from '@react-navigation/native-stack';
 import Animated, {
-  SharedTransition,
+  SharedElementTransition,
   withSpring,
+  SharedTransitionType,
 } from 'react-native-reanimated';
 
 const Stack = createNativeStackNavigator();
 
-const transition = SharedTransition.custom((values) => {
-  'worklet';
-  return {
-    width: withSpring(values.targetWidth),
-    height: withSpring(values.targetHeight),
-    originX: withSpring(values.targetOriginX),
-    originY: withSpring(values.targetOriginY),
-  };
-});
+const transition = new SharedElementTransition()
```

Comment:
not a big fan of this API. Btw: are we keeping things compatible, that is, is the old way of defining animation still going to work?

What I don't like about this API is that it uses a pattern that doesn't fit well here or are only partially applicable. Specifically, for builder pattern I wouldn't expect method called "animation" to setup an animation. I think for this purpose we can abandon this pattern completely and instead have something that's much simpler, e.g. a method that takes animation callback and optional progress animation callback, for example:

```
createCustomSharedTransition({ animationCallback: (values) => {}, animationWithProgressCallback: () => {})
```